File: examples/sync/backpack_public.py
"""Read Backpack public market data."""

import dcex


def main() -> None:
    """Run the Backpack public market-data example."""
    client = dcex.backpack(preload_product_table=False)

    markets = client.get_markets()
    print(markets)

    time_response = client.get_time()
    print(time_response)

    client.close()


if __name__ == "__main__":
    main()

File: crates/dcex/examples/backpack_public.rs
use std::time::Duration;

use dcex::exchanges::backpack::BackpackClient;

#[tokio::main]
async fn main() -> dcex::Result<()> {
    let client = BackpackClient::public(5_000, Duration::from_secs(10))?;
    let markets = client.get_markets().await?;
    println!("{}", markets.data);
    Ok(())
}
